drivers/tapdisk-nbdserver.c

```diff
@@ -440,250 +539,168 @@ receive_newstyle_options(td_nbdserver_client_t *client, int new_fd, bool no_zero
 	size_t n_options;
 	uint32_t opt_code;
 	uint32_t opt_len;
-	uint64_t opt_version;
-	uint64_t exportsize;
 	struct nbd_export_name_option_reply handshake_finish;
-	char *buf = NULL;
-	char *exportname = NULL;
-	int ret = 0;
 	td_nbdserver_t *server = client->server;
 
 	for (n_options = 0; n_options < MAX_OPTIONS; n_options++) {
+		if(receive_info(new_fd, (char *)&n_option, sizeof(n_option)) < 0)
+			goto fail;
 
-		if(receive_info(new_fd, (char *)&n_option, sizeof(n_option)) < 0){
-			return -1;	
-		}
-
-		opt_version = be64toh(n_option.version);
-		if (NBD_OPT_MAGIC != opt_version){
-			ERR("Bad NBD option version %" PRIx64
-			    ", expected %" PRIx64,
-                            opt_version, NBD_OPT_MAGIC);
-			ret = -1;	
-			goto done;
+		if (NBD_OPT_MAGIC != be64toh(n_option.version)) {
+			ERR("Bad NBD option version %" PRIx64 ", expected %" PRIx64,
+				be64toh(n_option.version), NBD_OPT_MAGIC);
+			goto fail;
 		}
-
 		opt_len = be32toh (n_option.optlen);
 		if (opt_len > MAX_REQUEST_SIZE) {
 			ERR ("NBD optlen to big (%" PRIu32 ")", opt_len);
-			ret = -1;	
-			goto done;
-		}
-
-		buf = malloc (opt_len + 1); 
-		if (buf == NULL) {
-			ERR("Could not malloc option data buff");
-			ret = -1;	
-			goto done;
+			goto fail;
 		}
-
 		opt_code = be32toh (n_option.option);
 
 		switch (opt_code) {
-		case NBD_OPT_EXPORT_NAME:
-		{
-			INFO("Processing NBD_OPT_EXPORT_NAME");
-			uint16_t flags = 0;
-			if(receive_info(new_fd, (char *)buf, opt_len) < 0) {
-				ERR ("Failed to received data for NBD_OPT_EXPORT_NAME");
-				ret = -1;
-				goto done;
-			}
-			buf[opt_len] = '\0';
-			INFO("Exportname %s", buf);
-
-			provide_server_info(server, &exportsize, &flags);
-
-			bzero(&handshake_finish, sizeof handshake_finish);
-			handshake_finish.exportsize = htobe64 (exportsize);
-      			handshake_finish.eflags = htobe16 (flags);
-			ssize_t len = no_zeroes ? 10 : sizeof(handshake_finish);
-			ssize_t sent = send (new_fd, &handshake_finish,len, 0);
-			if(sent != len) {
-				ERR ("Failed to send handshake finish");
-				ret = -1;	
-				goto done;
-			}
-		}
-		break;
-		case NBD_OPT_ABORT:
-			INFO("Processing NBD_OPT_ABORT");
-			if (send_option_reply (new_fd, opt_code, NBD_REP_ACK) == -1){
-				ret = -1;
+			case NBD_OPT_EXPORT_NAME:
+			{
+				char *buf = NULL;
+
+				INFO("Processing NBD_OPT_EXPORT_NAME");
+				if (!(buf=malloc(opt_len + 1))) { ERR("malloc fail"); goto fail; }
+				if(receive_info(new_fd, (char *)buf, opt_len) < 0) {
+					ERR ("Failed to received data for NBD_OPT_EXPORT_NAME");
+					free(buf);
+					goto fail;
+				}
+				buf[opt_len] = '\0';
+				INFO("Exportname %s", buf);
+
+				bzero(&handshake_finish, sizeof handshake_finish);
+				handshake_finish.exportsize = htobe64(NBD_EXPORTSIZE(server));
```

Comment:
Some inconsistent tabbing here?

Reply:
Ah yes. There was a mix of tabs and spaces around there originally, which confused matters. Fixed

Reply:
In some copious free time I need to run uncrustify over all this code once and for all.